cmd/kube-proxy/app/server.go

```diff
@@ -140,6 +141,7 @@ with the apiserver API to configure the proxy.`,
 		},
 	}
 
+	cmd.SetContext(genericapiserver.SetupSignalContext())
```

Comment:
if this is going to be used from kube-proxy and kube-scheduler then it seems like it shouldn't be in `k8s.io/apiserver`.